api/server/v1/marshaler.go

```diff
@@ -74,6 +80,9 @@ type Metadata struct {
 
 func CreateMDFromResponseMD(x *ResponseMetadata) Metadata {
 	var md Metadata
+	if x == nil {
```

Comment:
when can this happen? if this is happening then this is a bug.

Reply:
This is for tests on the client, to not fill metadata in every response.